server/src/gnome/secret_exchange.rs

```diff
@@ -0,0 +1,174 @@
+// SecretExchange: Exchange secrets between processes in an unexposed way.
+
+// Initial C implementation: https://gitlab.gnome.org/GNOME/gcr/-/blob/master/gcr/gcr-secret-exchange.c
+
+// The initial implementation of SecretExchange/GCRSecretExchange uses a KeyFile
+// to encode/parse the payload. In this implementation the payload is based
+// on a HashMap.
+// Before any transit operations the payload is base64 encoded and parsed into a
+// String.
+
+use std::collections::HashMap;
+
+use base64::prelude::*;
+use oo7::{crypto, Key};
+use zeroize::{Zeroize, Zeroizing};
+
+const SECRET: &str = "secret";
+const PUBLIC: &str = "public";
+const PRIVATE: &str = "private";
+const IV: &str = "iv";
+const PROTOCOL: &str = "[sx-aes-1]\n";
+const CIPHER_TEXT_LEN: usize = 16;
+
+#[derive(Debug, Zeroize)]
+pub struct SecretExchange {
```

Comment:
I am still not sure how the SecretExchange is supposed to be used. The current API, even if used internally only, is not really nice. why there is a need for a SecretExchange struct at all? who would call begin/create_shared_secret? and what about retrieve_secret? I would need to see how it is used in a real-case scenario to provide any meaningful feedback